Domain/forms

Rodrigosiq03 opened the pull request.

Rodrigosiq03 committed:
versao superior boto3, com suporte ao python3.13

Rodrigosiq03 committed:
Atualiza a lógica de repositórios para incluir STAGE.DOTENV em get_form_repo e get_image_repo

Rodrigosiq03 committed:
Refatora a classe Field e suas subclasses para melhorar a validação de tipos e parâmetros

Rodrigosiq03 committed:
Refatora a classe Form para simplificar a inicialização e melhorar a validação de atributos

Rodrigosiq03 committed:
Refatora a validação de tipos nos construtores das classes InformationField e suas subclasses

Rodrigosiq03 committed:
Refatora a classe Justification para melhorar a validação de tipos e simplificar a lógica de seleção

Rodrigosiq03 committed:
Altera o tipo de section_id de str para int na classe Section e ajusta a validação correspondente

Rodrigosiq03 committed:
Atualiza a enumeração FORM_STATUS para incluir os estados corretos e remove os desatualizados

Rodrigosiq03 committed:
Atualiza a enumeração PRIORITY para usar valores numéricos em vez de strings

Rodrigosiq03 committed:
Refatora a validação de parâmetros no método from_dynamo e ajusta a criação de campos para incluir label, order e help_text

Rodrigosiq03 committed:
Refatora a classe FormDynamoDTO para simplificar a inicialização e a conversão de dados, removendo campos desnecessários e ajustando a estrutura de dados.

Rodrigosiq03 committed:
Refatora o método from_request da classe InformationFieldDTO para melhorar a validação de parâmetros e a criação de instâncias de campos, garantindo que todos os campos necessários sejam verificados antes da inicialização.

Rodrigosiq03 committed:
Refatora a classe JustificationDTO para melhorar a validação de parâmetros e a estrutura de dados, removendo atributos desnecessários e ajustando a criação de instâncias de JustificationOption e SelectedJustification.

Rodrigosiq03 committed:
Altera o tipo de section_id de str para int na classe SectionDTO e ajusta a conversão nos métodos from_request e from_dynamo para garantir a consistência dos dados.

Rodrigosiq03 committed:
Atualiza os métodos update_form_status, cancel_form e complete_form na interface IFormRepository para incluir parâmetros de data de atualização e cancelamento, melhorando a rastreabilidade das operações.

Rodrigosiq03 committed:
Refatora a classe FormRepositoryMock para ajustar a inicialização de formulários, alterando tipos de dados e melhorando a estrutura, além de incluir parâmetros de data nas funções de atualização, cancelamento e conclusão.

Rodrigosiq03 committed:
Atualiza métodos de manipulação de formulários para incluir parâmetros de data de atualização e cancelamento, melhorando a rastreabilidade das operações.

GabrielGodoy01 commented on src/shared/domain/entities/form.py:
O campo "justification" não deve ser opcional.

Para todo formulário é obrigatório uma lista de opções selecionáveis.

GabrielGodoy01 replied:
O erro esta na documentação.

GabrielGodoy01 commented on src/shared/domain/entities/justification.py:
Obrigatoriamente uma "Justification" possui o campo "options".

Rodrigosiq03 committed:
Refatora a classe Form para melhorar a inicialização e compatibilidade de atributos, ajustando a validação de justificativas e observações, além de incluir novos parâmetros de compatibilidade.

Rodrigosiq03 committed:
Refatora classes de campo e justificativa para melhorar a inicialização e compatibilidade de atributos, incluindo novos parâmetros como placeholder e vinculação de justificativas. Atualiza o DTO de seção para garantir a consistência do tipo de section_id e ajusta o repositório mock para refletir mudanças de prioridade e novos parâmetros nas funções de conclusão de formulário.

Rodrigosiq03 committed:
Refactor form and section tests for improved clarity and consistency

- Updated test cases in test_form.py to use the new Form constructor and validation logic.
- Changed section_id from string to integer in test_section.py for consistency.
- Refactored justification handling in test cases to use SelectedJustification.
- Adjusted field definitions to align with the latest entity structure.
- Enhanced error handling tests to cover new validation scenarios.
- Updated repository mock tests to reflect changes in Form and Section entities.

GabrielGodoy01 commented on src/shared/infra/repositories/form_repository_dynamo.py:
Analisar estrutura do banco de dados (PK, SK, GSIs)
